src/main/java/seedu/teachstack/model/person/Person.java

```diff
@@ -73,7 +73,8 @@ public boolean isSamePerson(Person otherPerson) {
         }
 
         return otherPerson != null
-                && otherPerson.getStudentId().equals(getStudentId());
+                && otherPerson.getStudentId().equals(getStudentId())
+                && otherPerson.isSameEmail(this);
```

Comment:
Do you think this should be `||` so that students with the same email but different ID will also be considered as the same person (ie. no different person can have the same email)

Reply:
I thought that such students (same email, different ID) would never exist because email must be unique?

Reply:
I see, as such do you think the operator should be or to avoid entry of such data

Reply:
Yes, I'll make that change